cmd/nginx-ingress/main.go

```diff
@@ -746,3 +754,29 @@ func processConfigMaps(kubeClient *kubernetes.Clientset, cfgParams *configs.Conf
 	}
 	return cfgParams
 }
+
+func updateSelfWithVersionInfo(kubeClient *kubernetes.Clientset, version string, nginxVersion string) {
+	pod, err := kubeClient.CoreV1().Pods(os.Getenv("POD_NAMESPACE")).Get(context.TODO(), os.Getenv("POD_NAME"), meta_v1.GetOptions{})
+	if err != nil {
+		glog.Errorf("Error getting pod: %v", err)
+		return
+	}
+
+	// Copy pod and update the annotations.
+	newPod := pod.DeepCopy()
+	ann := newPod.ObjectMeta.Annotations
+	if ann == nil {
+		ann = make(map[string]string)
+	}
+	ann[nginxVersionAnnotation] = strings.Split(nginxVersion, ": ")[1]
```

Comment:
I think we might want to split on `/` to just get the version for example `1.23.2` instead of `nginx/1.23.2`?

Reply:
I don't see the harm in including whether or not the NGINX installed is plus or not in the annotation - it's very clear then with a glance which one you're dealing with without having to check deployment flags, or cross-referencing which version corresponds with Plus.

Reply:
oh yeah I wasn't saying to remove plus, the version string looks like:
`nginx version: nginx/1.23.2 (nginx-plus-r28)`
and right we're just getting everything after `:`
`nginx/1.23.2 (nginx-plus-r28)`
I thought it would look cleaner to get everything after `/`
`1.23.2 (nginx-plus-r28)`
but it's just a nitpick 🙂

Reply:
Ahhh, my apologies, I misinterpreted your comment completely, I should have read it properly, I thought you were saying to just keep the numbers 🤦🏼‍♀️ That makes more sense!